crates/api/src/kitsune.rs

```diff
@@ -57,6 +57,15 @@ pub type DynKitsuneHandler = Arc<dyn KitsuneHandler>;
 
 /// The top-level Kitsune2 api trait.
 pub trait Kitsune: 'static + Send + Sync + std::fmt::Debug {
+    /// Register the kitsune handler. This should be called exactly once,
+    /// before any other apis are invoked on the instance. The only reason
+    /// this isn't in the factory is to support a struct containing a
```

Comment:
It took me a while to understand what you mean by this comment. If my interpretation is correct, it could potentially be rephrased to:
"The only reason this isn't in the factory is to support structs that contain a kitsune instance in one of their fields while at the same time acting as that instance's handler"

Reply:
Another way to describe this would be to think of this as a dependency injection system where you want a circular dependency between two objects. The dependency injection system can't create one or the other first, so you have to have a one way dependency and then inform the object about the reverse dependency after creation.

Not sure if that's helpful, it makes sense to me that way because I'm used to working with dependency injection so that's the simplest way to think about the way this is being set up.

Reply:
Awesome, thanks for the feedback! Let me know how this sounds: https://github.com/holochain/kitsune2/pull/135/commits/6c5610ff0a986bcb00aaaab26f26c3438b8db4cc

Reply:
sounds good, thank you :+1: